gitops/generate_build.go

```diff
@@ -232,10 +233,13 @@ func getParamsForComponentBuild(component appstudiov1alpha1.Component, isInitial
 	sourceCode := component.Spec.Source.GitSource.URL
 	outputImage := component.Spec.ContainerImage
 	var err error
+
+	if err = protectDefaultImageRepo(outputImage, component.Namespace); err != nil {
+		return []tektonapi.Param{}, err
```

Comment:
I just noticed that we don't check the error value returned from this function in [GenerateInitialBuildPipelineRun function](https://github.com/redhat-appstudio/application-service/blob/632e5372c62f42ac0e31625fec32f7022c36e520/gitops/generate_build.go#L98), which will result in ignoring the error (for example in case the `.spec.containerImage` is set to `quay.io/redhat-appstudio/user-workload`) and initiating the build anyway.

@Michkov do you think we should update that?

Reply:
yeah either we update https://github.com/redhat-appstudio/application-service/blob/632e5372c62f42ac0e31625fec32f7022c36e520/gitops/generate_build.go#L98 to no longer ignore the error, or we move the call to `protectDefaultImageRepo` here after the is initial build check immediately below

unless there is another consideration @psturc and I were missing when he and I discussed his finding

Reply:
updated, now returning error from GenerateInitialBuildPipelineRun

Reply:
LGTM 👍